src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -730,7 +730,15 @@ public void DeleteSecretKey(TransactionOperationContext context, string name)
 
         public Task<(long Etag, object Result)> DeleteDatabaseAsync(string db, bool hardDelete, string[] fromNodes)
         {
-            var deleteCommand = new DeleteDatabaseCommand(db)
+            var nodes = new List<string>();
+
+            using (ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+            using (ctx.OpenReadTransaction())
+            {
+                nodes.AddRange(GetClusterTopology(ctx).AllNodes.Keys);
```

Comment:
This is actually opened up to a race condition. If between the time this run and the command executed stuff changed...